core/src/main/scala/org/apache/spark/status/AppStatusSource.scala

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.status
+
+import java.util.concurrent.atomic.AtomicLong
+
+import com.codahale.metrics.{Gauge, MetricRegistry}
+
+import org.apache.spark.SparkConf
+import org.apache.spark.internal.config.ConfigBuilder
+import org.apache.spark.metrics.source.Source
+
+private [spark] class JobDuration(val value: AtomicLong) extends Gauge[Long] {
+  override def getValue: Long = value.get()
+}
+
+private[spark] class AppStatusSource extends Source {
+
+  override val metricRegistry = new MetricRegistry()
+
+  override val sourceName = "appStatus"
+
+  val jobDuration = new JobDuration(new AtomicLong(0L))
+
+  // Duration of each job in milliseconds
+  val JOB_DURATION = metricRegistry
+    .register(MetricRegistry.name("jobDuration"), jobDuration)
+
+  val FAILED_STAGES = metricRegistry.counter(MetricRegistry.name("failedStages"))
+
+  val SKIPPED_STAGES = metricRegistry.counter(MetricRegistry.name("skippedStages"))
+
+  val COMPLETED_STAGES = metricRegistry.counter(MetricRegistry.name("completedStages"))
+
+  val SUCCEEDED_JOBS = metricRegistry.counter(MetricRegistry.name("succeededJobs"))
+
+  val FAILED_JOBS = metricRegistry.counter(MetricRegistry.name("failedJobs"))
+
+  val COMPLETED_TASKS = metricRegistry.counter(MetricRegistry.name("completedTasks"))
+
+  val FAILED_TASKS = metricRegistry.counter(MetricRegistry.name("failedTasks"))
+
+  val KILLED_TASKS = metricRegistry.counter(MetricRegistry.name("killedTasks"))
+
+  val SKIPPED_TASKS = metricRegistry.counter(MetricRegistry.name("skippedTasks"))
+
+  val BLACKLISTED_EXECUTORS = metricRegistry.counter(MetricRegistry.name("blackListedExecutors"))
```

Comment:
Given that most metrics are counters, I would add a small inline function that reduces some of the repetition here, e.g.
```scala
def getCounter(prefix: String, name: String): Counter =
  metricRegistry.counter(MetricRegistry.name(prefix, name))
```

Reply:
+1